traits/ctraits.c

```diff
@@ -942,24 +941,6 @@ has_traits_init ( PyObject * obj, PyObject * args, PyObject * kwds ) {
         Py_DECREF( value );
     }
 
-    /* Notify any interested monitors that a new object has been created: */
-    for ( i = 0, n = PyList_GET_SIZE( _HasTraits_monitors ); i < n; i++ ) {
-        value = PyList_GET_ITEM( _HasTraits_monitors, i );
```

Comment:
Note that apart from the obvious issue that a second call to `trait_monitor` always fails, there's a more subtle problem in this C code: the use of `GET_ITEM` here is unsafe, and given that we're calling arbitrary Python code on each iteration (the handler could be doing anything), it's not safe to assume that the list hasn't changed size while we're iterating over it.